components/compass_theme_provider/compass_theme_provider.tsx

```diff
@@ -1,11 +1,78 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
+import {blendColors, convertToRgb, rgbToHex, setAlpha} from '@mattermost/compass-components/shared/color-utils';
 import React, {useState, useEffect} from 'react';
+import {createGlobalStyle} from 'styled-components';
 
 import ThemeProvider, {lightTheme} from '@mattermost/compass-components/utilities/theme';
 
 import {Theme} from 'mattermost-redux/types/themes';
+import Constants from '../../utils/constants';
+import * as UserAgent from '../../utils/user_agent';
+
+const updateCodeTheme = (userTheme: string): void => {
+    let cssPath = '';
+    Constants.THEME_ELEMENTS.forEach((element) => {
+        if (element.id === 'codeTheme') {
+            element.themes?.forEach((theme) => {
+                if (userTheme === theme.id) {
+                    cssPath = theme.cssURL;
+                }
+            });
+        }
+    });
+
+    const link = document.querySelector<HTMLLinkElement>('link.code_theme');
+    if (link && cssPath !== link.href) {
+        // changeCss('code.hljs', 'visibility: hidden');
+
+        const xmlHTTP = new XMLHttpRequest();
+
+        xmlHTTP.open('GET', cssPath, true);
+        xmlHTTP.onload = function onLoad() {
+            link.href = cssPath;
+
+            if (UserAgent.isFirefox()) {
+                link.addEventListener('load', () => {
+                    // changeCss('code.hljs', 'visibility: visible');
+                }, {once: true});
+            } else {
+                // changeCss('code.hljs', 'visibility: visible');
+            }
+        };
+
+        xmlHTTP.send();
+    }
+};
+
+const changeColor = (colourIn: string, amt: number) => {
+    let hex = colourIn;
+    let lum = amt;
+
+    // validate hex string
+    hex = String(hex).replace(/[^0-9a-f]/gi, '');
+    if (hex.length < 6) {
+        hex = hex[0] + hex[0] + hex[1] + hex[1] + hex[2] + hex[2];
+    }
+    lum = lum || 0;
+
+    // convert to decimal and change luminosity
+    let rgb = '#';
+    let c;
+    let i;
+    for (i = 0; i < 3; i++) {
+        c = parseInt(hex.substr(i * 2, 2), 16);
+        c = Math.round(Math.min(Math.max(0, c + (c * lum)), 255)).toString(16);
+        rgb += ('00' + c).substr(c.length);
+    }
+
+    return rgb;
+};
+
+const stripRGBDeclarator = (color: string) => color.slice(4, -1);
```

Comment:
what's the use case for stripping the declarator? Is it possible to agree on one approach when using Rgb variables so we don't need these extra functions?

Reply:
the function I use for converting hex to rgb and it returns a full rgb string (`rgb(255, 255, 255)`) and since we only want the numbers i just strip the `rgb(` and closing `)` to get that.

There could be a better solution, but not without refactoring the compass component helper